### PR TITLE
Cleanup, reorganize structure, refactoring ThrowableRendererInterface

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/ErrorHandler.php [hunk 1]
```diff
@@ -94,25 +113,23 @@ public function handleThrowable(\Throwable $t): void
 
     /**
      * Register this error handler.
+     *
+     * @param bool $exposeDetails
      */
-    public function register(): void
+    public function register(bool $exposeDetails = false): void
```
Comment (samdark) [inline src/ErrorHandler.php:119]: You can use ErrorHandler via `ErrorCatcher` only without registering it as PHP handler via calling `register()`. In this case there's no way to set `exposeDetails`.

Comment (samdark) [reply src/ErrorHandler.php:119]: I think there should be a dedicated method for it as it was before and `register()` should not have a side effect of changing `expose` setting.

File: templates/_call-stack-item.php [hunk 2]
```diff
@@ -2,40 +2,28 @@
...
-IDE
-<svg class="icon icon--new-window" focusable="false" aria-hidden="true" width="16" height="16">
-    <use href="#new-window"></use>
-</svg>
-HTML;
+/* @var $this \Yiisoft\ErrorHandler\Renderer\HtmlRenderer */
 ?>
-<li class="<?= ($index === 1 || !$this->isCoreFile($file)) ? 'application' : '' ?> call-stack-item"
-    data-line="<?= (int) ($line - $begin) ?>">
+<li class="<?=!empty($lines) ? 'application' : '' ?> call-stack-item" data-line="<?= (int) ($line - $begin) ?>">
```
Comment (samdark) [inline templates/_call-stack-item.php:13]: I think we can still check for `isCoreFile()` somehow such as by checking for vendor/yiisoft in file path.

File: templates/_call-stack-item.php [hunk 3]
```diff
@@ -2,40 +2,28 @@
...
     <div class="element-wrap">
         <div class="flex-1">
             <?php if ($file !== null): ?>
                 <span class="file-name">
-                    <?= sprintf('%d. %s', (int) $index, 'in ' . $this->htmlEncode($file))  ?>
+                    <?= "{$index}. in {$this->htmlEncode($file)}" ?>
                 </span>
             <?php endif ?>
 
-            <?php if ($this->traceLine !== '{html}') : ?>
```
Comment (samdark) [inline templates/_call-stack-item.php:30]: It was previously used to have an ability to use custom template. A typical use-case was creating IDE-specific links for when trace file is clicked, it is opened right in the IDE.

File: templates/development.php [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 /* @var $throwable \Throwable */
-/* @var $this \Yiisoft\ErrorHandler\HtmlRenderer */
+/* @var $request \Psr\Http\Message\ServerRequestInterface|null */
+/* @var $this \Yiisoft\ErrorHandler\Renderer\HtmlRenderer */
 
 use Yiisoft\FriendlyException\FriendlyExceptionInterface;
 
-$theme = $_COOKIE['yii-exception-theme'] ?? '';
-
+if ($request && isset($request->getCookieParams()['yii-exception-theme'])) {
```
Comment (samdark) [inline templates/development.php:8]: We can't get cookie in such way if the error happens in the config during DI container initialization. I think we still should use `$_COOKIE['yii-exception-theme']` as fallback.

**Review by samdark — CHANGES_REQUESTED** [review]

File: config/params.php [hunk 5]
```diff
@@ -1,20 +0,0 @@
-<?php
-
-declare(strict_types=1);
-return [
```
Comment (samdark) [inline config/params.php:4]: Why removing it?

File: templates/_call-stack-item.php (hunk 2 above)
Comment (devanych) [reply templates/_call-stack-item.php:13]: I don't understand why this is necessary, this check decides whether to disclose the call stack content or not. Therefore, checking the existence of the content is the most logical.

File: config/params.php (hunk 5 above)
Comment (devanych) [reply config/params.php:4]: Because it is not necessary and these parameters are not correct at all.

File: templates/_call-stack-item.php (hunk 2 above)
Comment (samdark) [reply templates/_call-stack-item.php:13]: The idea is that framework is well-covered with tests so displaying trace from parts of the framework will less likely help to solve the problem. The problem is usually in the application itself so we make default state to focus on these.

File: config/params.php (hunk 5 above)
Comment (samdark) [reply config/params.php:4]: OK...

File: templates/_call-stack-item.php (hunk 2 above)
Comment (devanych) [reply templates/_call-stack-item.php:13]: OK

File: templates/_call-stack-item.php (hunk 3 above)
Comment (devanych) [reply templates/_call-stack-item.php:30]: Returned, renamed the property to the `$traceHeaderLine`.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**